sync/src/tests/helpers.rs

```diff
@@ -378,8 +502,32 @@ impl<P> TestNet<P> where P: Peer {
 		}
 	}
 
+	pub fn deliver_io_messages(&mut self) {
+		for peer in 0..self.peers.len() {
+			if !self.peers[peer].is_io_queue_empty() {
```

Comment:
Since we always have a pattern of `is_*_empty` followed by `prune_*` maybe it's better to allow calls to `prune_io_messages` and do the check internally?

That would make the trait much leaner.